internal/service/agent/handler.go

```diff
@@ -44,7 +45,8 @@ func ValidateEnrollmentAccessFromContext(ctx context.Context, log logrus.FieldLo
 	if !ok {
 		return errors.New("no common name in certificate")
 	}
-	if cn != crypto.ClientBootstrapCommonName {
+	if cn != crypto.ClientBootstrapCommonName &&
+		!strings.HasPrefix(cn, crypto.ClientBootstrapCommonNamePrefix) {
 		log.Warningf("an attempt to perform enrollment with a certificate with CN %q has been detected", cn)
```

Comment:
should this be a log.Error? just looks a little strange that we are returning an error and warning log.